soc/espressif/Kconfig.ulp

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2025 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+DT_LPCORE_PARTITION := $(dt_nodelabel_path,lpcore_partition)
+
+menu "Ultra Low Power (ULP) Co-processor"
+	config ULP_COPROC_ENABLED
+		bool "Ultra Low Power (ULP) Co-processor"
+		default n
+		help
+		  Enable this feature if you plan to use the ULP Co-processor.
+		  Once this option is enabled, further ULP co-processor configuration will appear in the menu.
+
+	choice ULP_COPROC_TYPE
+		prompt "ULP Co-processor type"
+		depends on ULP_COPROC_ENABLED
+		default ULP_COPROC_TYPE_LP_CORE if SOC_SERIES_ESP32C6
+		help
+		  Choose the ULP Coprocessor type: ULP FSM (Finite State Machine) or ULP RISC-V.
+
+	config ULP_COPROC_TYPE_FSM
+		bool "ULP FSM (Finite State Machine)"
+		depends on SOC_SERIES_ESP32 || SOC_SERIES_ESP32S2 || SOC_SERIES_ESP32S3
+	config ULP_COPROC_TYPE_RISCV
+		bool "ULP RISC-V"
+		depends on SOC_SERIES_ESP32S2 || SOC_SERIES_ESP32S3
+	config ULP_COPROC_TYPE_LP_CORE
+		bool "LP core RISC-V"
+		depends on SOC_SERIES_ESP32C6
+	endchoice
+
+	config ULP_COPROC_RESERVE_MEM
+		int
+		prompt "RTC slow memory reserved for coprocessor"
+		depends on ULP_COPROC_ENABLED
+		default $(dt_node_reg_size_int,$(DT_LPCORE_PARTITION))
+		range 32 8176 if SOC_SERIES_ESP32 || SOC_SERIES_ESP32S2 || SOC_SERIES_ESP32S3
+		range 32 16352 if SOC_SERIES_ESP32C6
+		help
+		  Bytes of memory to reserve for ULP Co-processor firmware & data.
+		  Data is reserved at the beginning of RTC slow memory.
```

Comment:
not sure why this is Kconfig, it's dts and comes from dts, just use the dts value directly?